addon/components/x-form.js

```diff
@@ -71,18 +71,20 @@ export default Ember.Component.extend({
 
   init() {
     this._super(...arguments);
-    this._setUpChangeset();
+    if(!this.get('data')) {
+      throw new Error('x-form needs data');
+    }
```

Comment:
@cherewaty confirmed verbally that "newness" is handled outside of x-form, and that we'll maybe make that a new feature later on.